### PR TITLE
spec: more fixes based on Fedora review

### PR DESCRIPTION
**Pull request by siteshwar**
Related: https://bugzilla.redhat.com/show_bug.cgi?id=2275304

### CODE REVIEW

File: osh.spec [hunk 1]
```diff
@@ -234,19 +234,40 @@ fi
...
 %ghost %attr(640,root,apache) %{_sharedstatedir}/osh/hub/secret_key
 %ghost %attr(640,root,apache) %{_sysconfdir}/osh/hub/secrets/jira_secret
 %ghost %attr(640,root,apache) %{_sysconfdir}/osh/hub/secrets/bugzilla_secret
 
+%exclude %{python3_sitelib}/osh/hub/static/admin/css/vendor/select2/LICENSE-SELECT2.md
+%exclude %{python3_sitelib}/osh/hub/static/admin/fonts/LICENSE.txt
+%exclude %{python3_sitelib}/osh/hub/static/admin/img/LICENSE
+%exclude %{python3_sitelib}/osh/hub/static/admin/js/vendor/jquery/LICENSE.txt
+%exclude %{python3_sitelib}/osh/hub/static/admin/js/vendor/select2/LICENSE.md
+%exclude %{python3_sitelib}/osh/hub/static/admin/js/vendor/xregexp/LICENSE.txt
```
Comment (kdudka) [inline osh.spec:261]: Are all the `%exclude` lines really needed? I have never seen this in any other package.

File: osh/hub/scripts/osh-xmlrpc-client.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
```
Comment (kdudka) [inline osh/hub/scripts/osh-xmlrpc-client.py:1]: Why is this needed?  For example `osh/hub/manage.py` still contains `#!/usr/bin/env python3`.

File: setup.py [hunk 3]
```diff
@@ -39,6 +39,9 @@
         "osh/hub/scripts/osh-stats",
         "osh/worker/osh-worker",
     ],
+    "/usr/share/licenses/osh": [
```
Comment (kdudka) [inline setup.py:42]: I do not think we need this in `setup.py`.  You can install the license file in the spec file directly.  See e.g. `curl.spec`: https://src.fedoraproject.org/rpms/curl/blob/f9311ae69d7c143fec8f3282907ac95546869cfc/f/curl.spec#_378

File: osh.spec [hunk 4]
```diff
@@ -234,19 +234,40 @@ fi
 %{_sbindir}/osh-retention
 %{_sbindir}/osh-stats
 %{_sysconfdir}/osh/hub
+%exclude %{python3_sitelib}/osh/hub/scripts/osh-xmlrpc-client.py
+%exclude %{python3_sitelib}/osh/hub/scripts/umb-emit.py
+%attr(755,root,root) %{python3_sitelib}/osh/hub/scripts/osh-xmlrpc-client.py
+%attr(755,root,root) %{python3_sitelib}/osh/hub/scripts/umb-emit.py
```
Comment (kdudka) [inline osh.spec:240]: I do not think we have ever used these scripts from an installed RPM package.  I would just remove them in `%install`.

**Review by kdudka — CHANGES_REQUESTED** [review]

File: osh.spec (hunk 1 above)
Comment (siteshwar) [reply osh.spec:261]: These files are listed twice in `rpmbuild` output.

File: osh/hub/scripts/osh-xmlrpc-client.py (hunk 2 above)
Comment (siteshwar) [reply osh/hub/scripts/osh-xmlrpc-client.py:1]: https://docs.fedoraproject.org/en-US/packaging-guidelines/#_shebang_lines

File: setup.py (hunk 3 above)
Comment (siteshwar) [reply setup.py:42]: Shall I leave it as it is, if somebody may choose to have a non-rpm package for this in the future?

File: osh/hub/scripts/osh-xmlrpc-client.py (hunk 2 above)
Comment (kdudka) [reply osh/hub/scripts/osh-xmlrpc-client.py:1]: If you remove the scripts in `%install`, you do not need these changes.  `manage.py` should not removed though.

File: setup.py (hunk 3 above)
Comment (kdudka) [reply setup.py:42]: I guess that non-rpm packages have their own ways to handle license files.  This looks rather Fedora specific.

File: osh.spec [hunk 5]
```diff
@@ -234,15 +234,19 @@ fi
...
 %{_sysconfdir}/osh/hub
+%exclude %{python3_sitelib}/osh/hub/scripts/osh-xmlrpc-client.py
+%exclude %{python3_sitelib}/osh/hub/scripts/umb-emit.py
 %{python3_sitelib}/osh/hub
 %{_unitdir}/osh-retention.*
 %{_unitdir}/osh-stats.*
 %exclude %{python3_sitelib}/osh/hub/settings_local.py*
 %exclude %{python3_sitelib}/osh/hub/__pycache__/settings_local.*
+# These file should be readable and writable by respective groups.
 %dir %attr(775,root,root) %{_localstatedir}/log/osh
```
Comment (kdudka) [inline osh.spec:245]: The mode needs to be `755` to avoid conflicts with `osh-worker`:
```
Error: Transaction test error:
  file /var/log/osh conflicts between attempted installs of osh-worker-1.0.0.20240423.145427.g5c09e40.pr_261-1.el9.noarch and osh-hub-1.0.0.20240423.145427.g5c09e40.pr_261-1.el9.noarch
```

File: osh.spec [hunk 6]
```diff
@@ -205,12 +204,13 @@ install -D {osh/hub,%{buildroot}%{_sysconfdir}/httpd/conf.d}/osh-hub-httpd.conf
...
 %dir %{_sharedstatedir}/osh
+%license LICENSE
 
 %files worker
 %{python3_sitelib}/osh/worker
 %{_unitdir}/osh-worker.service
-%attr(754,root,root) %{_sbindir}/osh-worker
-%dir %attr(775,root,root) %{_localstatedir}/log/osh
+%attr(755,root,root) %{_sbindir}/osh-worker
+%dir %attr(755,root,root) %{_localstatedir}/log/osh
```
Comment (xsuchy) [inline osh.spec:213]: This is a standard attribute for the directory; you can safely remove it.

File: osh.spec [hunk 7]
```diff
@@ -276,9 +280,9 @@ pg_isready -h localhost && %{python3_sitelib}/osh/hub/manage.py migrate
 %systemd_postun osh-{retention,stats}.{service,timer}
 
 %files hub-conf-devel
-%attr(640,root,apache) %config(noreplace) %{python3_sitelib}/osh/hub/settings_local.py
-%attr(640,root,apache) %config(noreplace) %{python3_sitelib}/osh/hub/__pycache__/settings_local*.pyc
-%attr(640,root,apache) %config(noreplace) %{_sysconfdir}/httpd/conf.d/osh-hub-httpd.conf
+%attr(644,root,apache) %config(noreplace) %{python3_sitelib}/osh/hub/settings_local.py
+%attr(644,root,apache) %config(noreplace) %{python3_sitelib}/osh/hub/__pycache__/settings_local*.pyc
+%attr(644,root,apache) %config(noreplace) %{_sysconfdir}/httpd/conf.d/osh-hub-httpd.conf
```
Comment (xsuchy) [inline osh.spec:285]: You can simplify this by:
```
%files hub-conf-devel
%defattr(644, root, apache)
%config(noreplace) %{python3_sitelib}/osh/hub/settings_local.py
...
```

File: osh.spec [hunk 8]
```diff
@@ -234,15 +234,19 @@ fi
...
 %{python3_sitelib}/osh/hub
 %{_unitdir}/osh-retention.*
 %{_unitdir}/osh-stats.*
 %exclude %{python3_sitelib}/osh/hub/settings_local.py*
 %exclude %{python3_sitelib}/osh/hub/__pycache__/settings_local.*
+# These file should be readable and writable by respective groups.
 %dir %attr(775,root,root) %{_localstatedir}/log/osh
 %dir %attr(775,root,apache) %{_localstatedir}/log/osh/hub
-%ghost %attr(640,apache,apache) %{_localstatedir}/log/osh/hub/hub.log
 %attr(775,root,apache) %{_sharedstatedir}/osh/hub
```
Comment (xsuchy) [inline osh.spec:247]: Here you can utilize defattr too:
```
%defattr(644, root, apache)
```
And everything below this line will have this owner and attributes.

File: osh.spec [hunk 9]
```diff
@@ -234,15 +234,19 @@ fi
 %{_sbindir}/osh-retention
 %{_sbindir}/osh-stats
 %{_sysconfdir}/osh/hub
+%exclude %{python3_sitelib}/osh/hub/scripts/osh-xmlrpc-client.py
```
Comment (xsuchy) [inline osh.spec:237]: Note that you do not need to exclude this file. What I pointed is the error "E: non-executable-script". This means that the file has `#!...` but not executable bit. This is a confusing state.
The solution is **either** set executable bit **or** remove the first line with `#!`.

Comment (kdudka) [reply osh.spec:237]: True.  My proposal was to remove them completely because we always use the scripts from a git repository.  So they do not need to be packaged at all: https://github.com/openscanhub/openscanhub/pull/261#discussion_r1576198499

**Comment by xsuchy** [comment]
LGTM.
Can you merge it, build src.rpm and update BZ? I will let `fedora-review` tool do another round of checks.

**Comment by siteshwar** [comment]
> LGTM. Can you merge it, build src.rpm and update BZ? I will let `fedora-review` tool do another round of checks.

Thanks! Would it be possible for you to do a review directly here? That would make it easier for me to make changes:

SPEC: https://download.copr.fedorainfracloud.org/results/packit/openscanhub-openscanhub-261/fedora-rawhide-x86_64/07344116-osh/osh.spec
SRPM: https://download.copr.fedorainfracloud.org/results/packit/openscanhub-openscanhub-261/fedora-rawhide-x86_64/07344116-osh/osh-1.0.0.20240424.105826.gae7fde7.pr_261-1.fc41.src.rpm

**Review by kdudka — APPROVED** [review]

**Comment by siteshwar** [comment]
I am merging this for now. Let's follow up on the bugzilla for further reviews.
